packages-exp/auth-exp/src/core/strategies/custom_token.ts

```diff
@@ -96,8 +97,9 @@ export function setCustomTokenProvider(
       token,
       returnSecureToken: true
     });
+    const responseToken = _parseToken(response.idToken!);
     _assert(
-      response.localId === authInternal.currentUser?.uid,
+      responseToken && responseToken.sub === authInternal.currentUser?.uid,
```

Comment:
You can do `responseToken?.sub === authInternal.currentUser?.uid`

Reply:
Oof, good call, done!